File: sdk/python/src/agentcc/types/chat/chat_completion_chunk.py
"""ChatCompletionChunk — the streaming response model."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel, ConfigDict

from agentcc.types.shared import Usage


class FunctionCallDelta(BaseModel):
    name: str | None = None
    arguments: str | None = None


class ToolCallDelta(BaseModel):
    index: int
    id: str | None = None
    type: str | None = None
    function: FunctionCallDelta | None = None


class Delta(BaseModel):
    role: str | None = None
    content: str | None = None
    tool_calls: list[ToolCallDelta] | None = None


class StreamChoice(BaseModel):
    model_config = ConfigDict(extra="allow")

    index: int
    delta: Delta
    finish_reason: str | None = None
    logprobs: Any = None


class ChatCompletionChunk(BaseModel):
    model_config = ConfigDict(extra="allow")

    id: str
    object: str
    created: int
    model: str
    choices: list[StreamChoice]
    usage: Usage | None = None
    system_fingerprint: str | None = None

File: sdk/typescript/src/types/chat/chat-completion-chunk.ts
import type { Usage, AgentCCMetadata } from "../shared.js";

export interface FunctionCallDelta {
  name?: string;
  arguments?: string;
}

export interface ToolCallDelta {
  index: number;
  id?: string;
  type?: "function";
  function?: FunctionCallDelta;
}

export interface Delta {
  role?: string;
  content?: string | null;
  tool_calls?: ToolCallDelta[];
  refusal?: string | null;
}

export interface StreamChoice {
  index: number;
  delta: Delta;
  finish_reason: string | null;
  logprobs?: unknown;
}

export interface ChatCompletionChunk {
  id: string;
  object: "chat.completion.chunk";
  created: number;
  model: string;
  choices: StreamChoice[];
  usage?: Usage;
  system_fingerprint?: string;
  /** Gateway metadata — injected by the SDK from response headers. */
  agentcc?: AgentCCMetadata;
  [key: string]: unknown;
}
